1. Print out a data type of a variable?

In [13]:
example = "3.1415926536"
type(example)

str

2. Round 4.5667 to the nearest hundreth using a build-in function, round().

In [2]:
round(4.5667, 2)

4.57

3. Convert "657" to an integer data type

In [1]:
int("657")

657

4. Write a function to calculate the miles per gallon.
Get a miles driven and gallons used from a user and call your function to calculate the miles per gallon. 

In [15]:
def miles_per_gallon (miles_driven, gallons_used):
    MPG = miles / gallon
    print("Miles Per Gallon:", MPG)

    
miles = int(input("Enter miles driven: "))
gallon = int(input("Enter gallons of gas used: "))

miles_per_gallon(miles, gallon)

Enter miles driven: 150
Enter gallons of gas used: 15
Miles Per Gallon: 10.0


5. Enhance your function from exercise 3. Additionally, get a gas price per gallon from a user and write a program to compute the total gas cost and cost per mile. 

The output should look something like this:

    Enter miles driven:            150
    Enter gallons of gas used:     15
    Enter cost per gallon:          3

    Miles Per Gallon:   10.0
    Total Gas Cost:     45.0
    Cost per Mile:      0.3


In [14]:
def road_trip (miles_driven, gallons_used, cost_per_gallon):
    MPG = miles_driven / gallons_used
    cost_gas_total = gallons_used * cost_per_gallon 
    cost_per_mile = cost_gas_total / miles_driven

    print(f"Miles Per Gallon: \t {MPG}")
    print(f"Total Gas Cost: \t {cost_gas_total}")
    print(f"Cost per Mile: \t\t {cost_per_mile}")


miles = int(input("Enter miles driven: \t\t"))
gallon = int(input("Enter gallons of gas used: \t"))
cost = int(input("Enter cost per gallon: \t\t"))

road_trip(miles, gallon, cost)

Enter miles driven: 		150
Enter gallons of gas used: 	15
Enter cost per gallon: 		3
Miles Per Gallon: 	 10.0
Total Gas Cost: 	 45
Cost per Mile: 		 0.3


6. Write a function called multiply_two() that accepts two parameters. The function should print a message like '15 + 2 = 17'.  Randomly generate two numbers between 1 and 100, then call the function with the two random numbers.

In [27]:
import random as rand

def multiply_two(n1, n2):
    print(f'{n1} + {n2} = {n1 + n2}')
    
num1 = rand.randint(1,100) 
num2 = rand.randint(1,100) 

multiply_two(num1, num2)

37 + 50 = 87


7. Write a function called make_shirt() that accepts a size and the text of a message. The function should print a sentence summarizing the size of the shirt and the message printed on it. The default shirt size is large and default message is 'I love Python!' Make a default size shirt with a message, a medium shirt with the default message, and a shirt of any size with a different message.

In [4]:
def make_shirt(size='large', message='I love Python!'):
    print("Shirt size:", size)
    print("Shirt says:", message, "\n") # added line break
    
make_shirt() # default
make_shirt('medium')
make_shirt('any size', "Why it doesn't fit?")

Shirt size: large
Shirt says: I love Python! 

Shirt size: medium
Shirt says: I love Python! 

Shirt size: any size
Shirt says: Why it doesn't fit? 



8. Create a function for a multiplication table.  Get a number from a user and then call you function which produces a multiplication table for that number.
The output should look something like this:

        Enter a number: 3
        3 X 1 = 3
        3 X 2 = 6
        3 X 3 = 9
        3 X 4 = 12
        3 X 5 = 15
        3 X 6 = 18
        3 X 7 = 21
        3 X 8 = 24
        3 X 9 = 27
        

In [6]:
number = int(input("Enter a number: "))

print(f"{number} X 1 = {number * 1}")
print(f"{number} X 2 = {number * 2}")
print(f"{number} X 3 = {number * 3}")
print(f"{number} X 4 = {number * 4}")
print(f"{number} X 5 = {number * 5}")
print(f"{number} X 6 = {number * 6}")
print(f"{number} X 7 = {number * 7}")
print(f"{number} X 8 = {number * 8}")
print(f"{number} X 9 = {number * 9}")


Enter a number: 11
11 X 1 = 11
11 X 2 = 22
11 X 3 = 33
11 X 4 = 44
11 X 5 = 55
11 X 6 = 66
11 X 7 = 77
11 X 8 = 88
11 X 9 = 99


9. Write a function that accepts a string and returns a string that is 
    - without any leading nor trailing blanks
    - converts to all capital letters
    - and replaces all blanks with a period(.).  

Write a program to accept a string from a user then call your function and display the original string and changed string.  

The output should look something like this:

        Enter a text: may your dreams come true
        
        Original Text: may your dreams come true
        New Text: MAY.YOUR.DREAMS.COME.TRUE

In [7]:
def robot_speak (message):
    message = message.replace(" ", ".")
    message = message.upper()
    print(message)
    
text = input("Enter a text: ")

robot_speak (text)

Enter a text: this is a sentence
THIS.IS.A.SENTENCE


10. Write a function that returns a randomly selected character from a passed string. Write a program to accept a string from a user then call your function and display the original string and selected character.

The output should look something like this:

        Enter a text: functions
        Original Text:	functions
        Random character:	s

In [10]:
import random as rand

def random_char(input_text):
    select = rand.randint(0, len(input_text))
    print("Original Text:\t", input_text)
    print("Random character:\t", input_text[select])
    
text = input("Enter a text: ")

random_char(text)

Enter a text: supercalifragilisticexpialidocious
Original Text:	 supercalifragilisticexpialidocious
Random character:	 c
